contracts/extensions/bridge-operator-governance/BridgeManager.sol

```diff
@@ -0,0 +1,580 @@
+
+  /**
+   * @dev The numerator value used for calculations in the contract.
+   * @notice value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin.numerator.slot") - 1
+   */
+  TUint256Slot internal constant NUMERATOR_SLOT =
+    TUint256Slot.wrap(0xc55405a488814eaa0e2a685a0131142785b8d033d311c8c8244e34a7c12ca40f);
+
+  /**
+   * @dev The denominator value used for calculations in the contract.
+   * @notice value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin.denominator.slot") - 1
+   */
+  TUint256Slot internal constant DENOMINATOR_SLOT =
+    TUint256Slot.wrap(0xac1ff16a4f04f2a37a9ba5252a69baa100b460e517d1f8019c054a5ad698f9ff);
+
+  /**
+   * @dev The nonce value used for tracking nonces in the contract.
+   * @notice value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin.nonce.slot") - 1
+   */
+  TUint256Slot internal constant NONCE_SLOT =
+    TUint256Slot.wrap(0x92872d32822c9d44b36a2537d3e0d4c46fc4de1ce154ccfaed560a8a58445f1d);
+
+  /**
+   * @dev The total weight value used for storing the cumulative weight in the contract.
+   * @notice value is equal to keccak256("@ronin.dpos.gateway.BridgeAdmin.totalWeights.slot") - 1
+   */
+  TUint256Slot internal constant TOTAL_WEIGHTS_SLOT =
+    TUint256Slot.wrap(0x6924fe71b0c8b61aea02ca498b5f53b29bd95726278b1fe4eb791bb24a42644c);
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  bytes32 public immutable DOMAIN_SEPARATOR;
+
+  modifier onlyGovernor() virtual {
+    _requireGovernor(msg.sender);
+    _;
+  }
+
+  constructor(
+    uint256 num,
+    uint256 denom,
+    uint256 roninChainId,
+    address bridgeContract,
+    address[] memory callbackRegisters,
+    address[] memory bridgeOperators,
+    address[] memory governors,
+    uint96[] memory voteWeights
+  ) payable BridgeManagerCallbackRegister(callbackRegisters) {
+    NONCE_SLOT.store(1);
+    NUMERATOR_SLOT.store(num);
+    DENOMINATOR_SLOT.store(denom);
+
+    _setContract(ContractType.BRIDGE, bridgeContract);
+
+    DOMAIN_SEPARATOR = keccak256(
+      abi.encode(
+        keccak256("EIP712Domain(string name,string version,bytes32 salt)"),
+        keccak256("BridgeAdmin"), // name hash
+        keccak256("2"), // version hash
+        keccak256(abi.encode("BRIDGE_ADMIN", roninChainId)) // salt
+      )
+    );
+
+    _addBridgeOperators(voteWeights, governors, bridgeOperators);
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function addBridgeOperators(
+    uint96[] calldata voteWeights,
+    address[] calldata governors,
+    address[] calldata bridgeOperators
+  ) external onlySelfCall returns (bool[] memory addeds) {
+    addeds = _addBridgeOperators(voteWeights, governors, bridgeOperators);
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
+   */
+  function removeBridgeOperators(
+    address[] calldata bridgeOperators
+  ) external onlySelfCall returns (bool[] memory removeds) {
+    removeds = _removeBridgeOperators(bridgeOperators);
+  }
+
+  /**
+   * @inheritdoc IBridgeManager
```

Comment:
```suggestion
   * @inheritdoc IBridgeManager
   * @notice This method checks authorization by querying the corresponding operator of the msg.sender and then
   * attempts to remove it from the `_bridgeOperatorSet` for gas optimization. In case we allow a governor can leave
   * their operator address blank null `address(0)`, consider add authorization check.
```